python/pyspark/ml/feature.py

```diff
@@ -1110,6 +1110,80 @@ class PCAModel(JavaModel):
     """
 
 
+@inherit_doc
+class RFormula(JavaEstimator, HasFeaturesCol, HasLabelCol):
+    """
+    .. note:: Experimental
+
+    Implements the transforms required for fitting a dataset against an
+    R model formula. Currently we support a limited subset of the R
+    operators, including '~', '+', '-', and '.'. Also see the R formula
+    docs:
+    http://stat.ethz.ch/R-manual/R-patched/library/stats/html/formula.html
+
+    >>> df = sqlContext.createDataFrame([
+    ...     (1.0, 1.0, "a"),
+    ...     (0.0, 2.0, "b"),
+    ...     (0.0, 0.0, "a")
+    ... ], ["y", "x", "s"])
+    >>> rf = RFormula(formula="y ~ x + s")
+    >>> rf.fit(df).transform(df).show()
```

Comment:
Not a PySpark comment, but is this expected behavior?

     >>> df = sqlContext.createDataFrame([
    ...     (1.0, 1.0, "a"),
    ...     (0.0, 2.0, "b"),
    ...     (0.0, 0.0, "c")
    ... ], ["y", "x", "s"])
    >>> rf.fit(df).transform(df).collect()
    [Row(y=1.0, x=1.0, s='a', features=DenseVector([1.0, 1.0, 0.0]), label=1.0),
     Row(y=0.0, x=2.0, s='b', features=DenseVector([2.0, 0.0, 1.0]), label=0.0),
     Row(y=0.0, x=0.0, s='c', features=SparseVector(3, {}), label=0.0)]